### PR TITLE
[QUIC] QUIC packetezation

### PR DESCRIPTION
**Pull request by Mr3zee**
**Subsystem**
`ktor-network`

**Solution**
Handlers for QUIC packets read and write operations

### CODE REVIEW

File: ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/bytes/ByteUtils.kt [hunk 1]
```diff
@@ -2,17 +2,69 @@
...
+internal inline fun UInt8.toUInt16(): UInt16 = toUShort()
+
+internal inline fun UInt64.toUInt8(): UInt8 = toUByte()
+internal inline fun UInt32.toUInt8(): UInt8 = toUByte()
+internal inline fun UInt16.toUInt8(): UInt8 = toUByte()
+internal inline fun UInt64.toUInt16(): UInt16 = toUShort()
+internal inline fun UInt32.toUInt16(): UInt16 = toUShort()
+internal inline fun UInt64.toUInt32(): UInt32 = toUInt()
+
+@OptIn(ExperimentalUnsignedTypes::class)
```
Comment (e5l) [inline ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/bytes/ByteUtils.kt:31]: Could be replaced with file:
@file:OptIn(ExperimentalUnsignedTypes::class)

File: ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/consts/NumericConsts.kt [hunk 2]
```diff
@@ -2,10 +2,13 @@
  * Copyright 2014-2023 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
  */
 
-package io.ktor.network.quic.util
+package io.ktor.network.quic.consts
```
Comment (e5l) [inline ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/consts/NumericConsts.kt:5]: Could you tell me why a separate packet is needed? I would keep it in `quic`

**Review by e5l — APPROVED** [review]
some end-to-end tests would be great at some point

File: ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/bytes/ByteUtils.kt (hunk 1 above)
Comment (Mr3zee) [reply ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/bytes/ByteUtils.kt:31]: I'm not sure about compiling, but IDE shows me that this approach does not work (it still requires opt-in for separate functions)

File: ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/consts/NumericConsts.kt (hunk 2 above)
Comment (Mr3zee) [reply ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/consts/NumericConsts.kt:5]: I just that I like to keep everything in separate packages and not mix files, easier to navigate for me

Comment (Mr3zee) [reply ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/consts/NumericConsts.kt:5]: but maybe for this particular file `util` directory is indeed better

File: ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/bytes/ByteUtils.kt (hunk 1 above)
Comment (Mr3zee) [reply ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/bytes/ByteUtils.kt:31]: checked, guess it's IDE bug

**Comment by Mr3zee** [comment]
> some end-to-end tests would be great at some point

yes, as soon as I fill all stubs, I will cover this functionality with tests

**Pull request merged**